client/r/rhawc/R/rclientClass.R

```diff
@@ -0,0 +1,158 @@
+#' r client functions
+#'
+#' pretty please work
+#'
+#' @import httr
+#' @import plyr
+#' @export
+HawcClient <- function(baseUrl){
+
+  # local environment variables
+  thisEnv <- environment()
+
+  # persistent "state"
+  root_url <- baseUrl
+  #   token <- ""
+
+  checkIsAuthenticated <- function(){
+    if((nchar(get("token", thisEnv))==0)){
+      stop('Authentication required')
+    }
+    else if (get("token", thisEnv) == 'Unable to log in with provided credentials.') {
+      stop('Unable to log in with provided credentials.')
+    }
+  }
+  get_ = function(url) {
+    response = GET(url, add_headers(Authorization=paste('Token', get("token", thisEnv),sep=' '),
+                                    'Content-Type'="application/json"))
+    if (response$status_code >= 400 & response$status_code < 500) {
+      stop("An exception occured in the HAWC client module") }
+
+    else if (response$status_code == 500) {
+      stop("An exception occured on the HAWC server") }
+
+    else {
+      response.list = content(response)
```

Comment:
some responses should return a data frame, but not all.  in the python client, when it returns a Dict, here we should return the `response.list` and when it returns a datafarmae, we should return a dataframe. I'd make the get_ and post_ methods return a `response.list`, and then for some methods use that as.data.frame method to convert the list to a frame